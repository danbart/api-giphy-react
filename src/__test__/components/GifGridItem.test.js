import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import { GifGridItem } from "../../components/GifGridItem";

const title = "un titulo";
const url = "https://localhost/";

const wrapper = shallow(<GifGridItem title={title} url={url} />);

it("Renderizar correctamente", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("Debe de tener un parrafo con el titulo", () => {
  const p = wrapper.find("p");
  expect(p.text().trim()).toBe(title);
});

it("Debe de tener la imagen igual al url y alt de los props", () => {
  const img = wrapper.find("img");
  expect(img.prop("src")).toBe(url);
  expect(img.prop("alt")).toBe(title);
});

it("Debe de tener la clase animate__fadeIn", () => {
  const div = wrapper.find("div");
  expect(div.prop("className").includes("animate__fadeIn")).toBe(true);
});
