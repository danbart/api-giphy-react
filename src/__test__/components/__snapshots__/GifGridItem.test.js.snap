// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizar correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https://localhost/"
  />
  <p>
    un titulo
  </p>
</div>
`;
